### PR TITLE
Replace form modules with proper view module functions

### LINKED ISSUE #119: Form module build_and_merge merges arbitrary keys into structs
Structs in Elixir work because they have a fixed, predefined set of keys.  Adding arbitrary keys to the underlying map of the struct violates invariants for the struct and leads to really confusing errors about keys missing that shouldn't be in the struct.  Additionally, the form module pattern seems to be treating the struct like form objects, but in Phoenix, the helper functions should on the view module, whose functions are available from the templates.

- [ ] Don't merge maps with new keys into structs
- [ ] Move functions in form modules to view modules

### PR DESCRIPTION
**Pull request by KronicDeth**
Fixes #119 

# Changelog
## Enhancements
* Unit conversion for wei is moved down to `Explorer.Chain` and below.  `ExplorerWeb` is only responsible for formatting to a string.
* CircleCI now has `gettext` job, which runs `mix gettext.extract --merge` to check that the the `.pot` and `.po` files are up-to-date when there are any changes to `gettext` calls.  Developers should run `mix gettext.extract --merge` in `apps/explorer_web` to update the files and commit any changes.  **DO NOT** run `mix gettext.extract --merge` from the root of the project.  It leads to duplicate usage entries in the `.pot` and `.po` files due to how recursive compilation works.

## Bug Fixes
* `ExplorerWeb.BlockForm` is no longer used, so it no longer tries to inject new keys into the `%Explorer.Chain.Block{}`, instead the proper Phoenix idiom of adding functions to the view module, `ExplorerWeb.BlockView`, is done as they are exposed in the block templates.
* `ExplorerWeb.PendingTransactionForm` is no longer used, so it no longer tries to inject new keys into `%Explorer.Chain.Transaction{}`, instead the proper Phoenix idiom of adding functions to the view module, `ExplorerWeb.PendingTransactionView`, is done as they are exposed in the pending transaction templates.
* `ExplorerWeb.TransactionForm` is no longer used, so it no longer tries to inject new keys into `%Explorer.Chain.Transaction{}`, instead the proper Phoenix idiom of adding functions to the view module, `ExplorerWeb.TransactionView`, is done as they are exposed in the transaction templates.
* Wei and gas consistently use `Cldr.Number.to_string!` for both whole and fractional numbers.
  * Same formatting function is used from view modules, instead of calculating unit conversions in templates.  The actual unit conversion happens in `Explorer.Chain` as mentioned in enhancements above.
* `gettext "POA"` changed to `gettext "Ether"` as the generic unit is `Ether` while `POA` is chain-specific.  `POA` is still the default translation by setting it in `apps/explorer_web/priv/gettext/default.pot`.

## Incompatible Changes
* `ExplorerWeb.BlockForm` is removed.  Functions are moved to `ExplorerWeb.BlockView`.
* `ExplorerWeb.PendingTransactionForm` is removed.  Functions are moved to `ExplorerWeb.PendingTransactionForm`.
* `ExplorerWeb.TransactionForm` is removed.  Functions are moved to `ExplorerWeb.TransactionForm`.

### CODE REVIEW

**Comment by KronicDeth** [comment]
@acravenho, @rstormsf please check the new docs on `apps/explorer/lib/explorer/chain/*.ex` files to ensure they describe the domain correctly.

File: apps/explorer_web/lib/explorer_web/views/address_transaction_from_view.ex [hunk 1]
```diff
@@ -1,4 +1,8 @@
 defmodule ExplorerWeb.AddressTransactionFromView do
   use ExplorerWeb, :view
-  @dialyzer :no_match
+
+  alias ExplorerWeb.TransactionView
+
+  defdelegate status(transacton), to: TransactionView
```
Comment (alexgaribay) [inline apps/explorer_web/lib/explorer_web/views/address_transaction_from_view.ex]: I haven't seen `defdelegate` used before.

**Comment by tmecklem** [comment]
I would have liked a little more time to think this PR over. I agree that the Form "Object" approach wasn't a great pattern to use, especially with the extension of the structs. While the `defdelegate` approach of code reuse in the Views is much better than the struct extension approach, it still doesn't feel quite right yet.

I'm struggling to put into words what about it causes me concern, but I think the thing that gives me pause is the pattern of Views pointing to other views and the indirection that results. It's not clear to me where the function should be defined (where it lives) vs modules where it should be delegated. I would like a pattern establish where we have a clear picture of where these shared presentation concerns go and how they are invoked that isn't subject to my gut feeling that day.

@KronicDeth I'd love to know if you've thought of this already and have an idea, or if you're interested in exploring alternatives with us going forward once we have worked in this iteration of the code for a while.

**Comment by KronicDeth** [comment]
@tmecklem I'm fine with not doing the `defdelegate`.  It was mostly to keep the number formatting consistent because some places weren't using `Cldr.Number.to_string`.  I'm fine with duping that if you prefer it.  It may be also be redundant with moving to a combined value+unit type that @alexgaribay is now working on.

**Comment by tmecklem** [comment]
I'm very interested in the last thing you mentioned. Having that combined value and unit type might provide an elegant path. I'll check that out. Thanks, @KronicDeth.

**Pull request merged**